docarray/array/doc_list/io.py

```diff
@@ -361,40 +362,52 @@ def from_csv(
             'unix' (for csv file generated on UNIX systems).
         :return: DocList
         """
-        from docarray import DocList
-
         if cls.doc_type == AnyDoc:
             raise TypeError(
                 'There is no document schema defined. '
                 'Please specify the DocList\'s Document type using `DocList[MyDoc]`.'
             )
 
+        if file_path.startswith('http'):
+            import urllib.request
+
+            with urllib.request.urlopen(file_path) as f:
+                file_as_string = StringIO(f.read().decode(encoding))
+                rows = csv.DictReader(file_as_string, dialect=dialect)
+                docs = cls._from_csv_dict_reader(rows)
+        else:
+            with open(file_path, 'r', encoding=encoding) as fp:
+                rows = csv.DictReader(fp, dialect=dialect)
+                docs = cls._from_csv_dict_reader(rows)
```

Comment:
you can factorise your code so that the line csv.DictReader and from_csv_dict_reader is not repeated

Reply:
or maybe it is not that easy with the context manager ?

Reply:
you can create a function that you call inside both context